src/c++/perf_analyzer/test_perf_utils.cc

```diff
@@ -114,4 +117,288 @@ TEST_CASE("testing the ParseTensorFormat function")
   CHECK(ParseTensorFormat("") == cb::TensorFormat::UNKNOWN);
 }
 
+TEST_CASE("perf_utils: ParseProtocol")
+{
+  CHECK(ParseProtocol("HTTP") == cb::ProtocolType::HTTP);
+  CHECK(ParseProtocol("http") == cb::ProtocolType::HTTP);
+  CHECK(ParseProtocol("GRPC") == cb::ProtocolType::GRPC);
+  CHECK(ParseProtocol("grpc") == cb::ProtocolType::GRPC);
+  CHECK(ParseProtocol("hhttp") == cb::ProtocolType::UNKNOWN);
+  CHECK(ParseProtocol("") == cb::ProtocolType::UNKNOWN);
+  CHECK(ParseProtocol("http2") == cb::ProtocolType::UNKNOWN);
+}
+
+TEST_CASE("perf_utils: ConvertDTypeFromTFS")
+{
+  std::string datatype;
+  cb::Error status;
+
+  SUBCASE("Check for correct conversion")
+  {
+    status = ConvertDTypeFromTFS("DT_HALF", &datatype);
```

Comment:
any way to shorten this? A lambda? Or a list of pairs that you loop through?

Reply:
I didn't loop through them because I thought it would make them less readable as test cases but I could try to shorten them if the current version is too long.

Reply:
I'm a fan of verbose tests like this rather than loops. Loops are less readable imo for someone. Besides verifying behavior, unit tests are intended to teach people what the code is doing. So the more readable, the better.

Reply:
Changed to looping through pairs since I don't think it loses too much readability.